state/mongodb/mongodb.go

```diff
@@ -146,9 +148,18 @@ func (m *MongoDB) Init(metadata state.Metadata) error {
 
 	m.metadata = *meta
 	opts := options.Collection().SetWriteConcern(wc).SetReadConcern(rc)
-	collection := m.client.Database(meta.DatabaseName).Collection(meta.CollectionName, opts)
-
-	m.collection = collection
+	m.collection = m.client.Database(meta.DatabaseName).Collection(meta.CollectionName, opts)
+
+	// Set expireAfterSeconds index on ttl field with a value of 0 to delete
+	// values immediately when the TTL value is reached.
+	// MongoDB TTL Indexes: https://docs.mongodb.com/manual/core/index-ttl/
+	// TTL fields are deleted at most 60 seconds after the TTL value is reached.
+	if _, err := m.collection.Indexes().CreateOne(context.Background(), mongo.IndexModel{
```

Comment:
What happens if the index already exists? Is this a no-op? And what happens if the collection already exists and has a lot of data - will this block? And what if 2 instances of Dapr create the index at the same time?

Also, nit, but I find putting complex, multi-line function invocations inside an `if` to be really hard to read

Reply:
> What happens if the index already exists? Is this a no-op? 
no-op, unless the index already exists _and_ has different options, in which case it errors out. I feel like erroring out is the safest option? Maybe it doesn't matter since the index name is already namespaced with dapr (`_dapr_ttl`) ...

>  And what happens if the collection already exists and has a lot of data - will this block?
I have added `SetBackground(true)`.

`And what if 2 instances of Dapr create the index at the same time?`
Same as above, noop.

Reply:
Thanks for confirming :)

> unless the index already exists and has different options, in which case it errors out.

I am not worried about that. Users should not be modifying the schema of the collection outside of Dapr.